### DISCUSSION TITLE
.NET 6 hosting model

### DISCUSSION THREAD

**Discussion by fgheysels** [arcus-azure/arcus.templates#467] [General]
.NET 6 uses a new hosting model.  You can see that, when starting a new Web API project in VS.NET 2022, that the newly created project doesn't contain a `Startup` class anymore.  
The `Program.cs` file consists of top-level statements:

```csharp
var builder = WebApplication.CreateBuilder(args);

// Add services to the container.

builder.Services.AddControllers();
// Learn more about configuring Swagger/OpenAPI at https://aka.ms/aspnetcore/swashbuckle
builder.Services.AddEndpointsApiExplorer();
builder.Services.AddSwaggerGen();

var app = builder.Build();

// Configure the HTTP request pipeline.
if (app.Environment.IsDevelopment())
{
    app.UseSwagger();
    app.UseSwaggerUI();
}

app.UseHttpsRedirection();

app.UseAuthorization();

app.MapControllers();

app.Run();
```

As can be seen above, `HostBuilder` is no longer used (directly), instead `WebApplicationBuilder` is used.

Although the existing/old model that is used by .NET Core and .NET 5 is still supported by .NET 6, I feel that the Arcus Web API template must be modified so that the new hosting model is used.

More information can be found [here](https://gist.github.com/davidfowl/0e0372c3c1d895c3ce195ba983b1e03d).

**Comment by stijnmoreels** [comment]
Yes, I was thinking the same thing. Good one. 👍
